zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/ElasticsearchEngineClient.java

```diff
@@ -0,0 +1,183 @@
+    try {
+      client.indices().create(request);
+      log.debug("Index [{}] was successfully created", indexDescriptor.getIndexName());
+    } catch (final IOException e) {
+      log.error("Index [{}] was NOT created", indexDescriptor.getIndexName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void createIndexTemplate(final IndexTemplateDescriptor templateDescriptor) {
+    final PutIndexTemplateRequest request = putIndexTemplateRequest(templateDescriptor);
+
+    try {
+      client.indices().putIndexTemplate(request);
+      log.debug("Template [{}] was successfully created", templateDescriptor.getTemplateName());
+    } catch (final IOException e) {
+      log.error("Template [{}] was NOT created", templateDescriptor.getTemplateName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void createComponentTemplate(final ComponentTemplateDescriptor templateDescriptor) {
+    final PutComponentTemplateRequest request = putComponentTemplateRequest(templateDescriptor);
+
+    try {
+      client.cluster().putComponentTemplate(request);
+      log.debug(
+          "Component template [{}] was successfully created", templateDescriptor.getTemplateName());
+    } catch (final IOException e) {
+      log.error("Component template [{}] was NOT created", templateDescriptor.getTemplateName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void putMapping(final IndexDescriptor indexDescriptor, final String properties) {
+    final PutMappingRequest request = putMappingRequest(indexDescriptor, properties);
+
+    try {
+      client.indices().putMapping(request);
+      log.debug("Mapping in [{}] was successfully updated", indexDescriptor.getIndexName());
+    } catch (final IOException e) {
+      log.error("Mapping in [{}] was NOT updated", indexDescriptor.getIndexName(), e);
+      throw new ExporterElasticsearchException(e.getMessage());
+    }
+  }
+
+  private PutMappingRequest putMappingRequest(
+      final IndexDescriptor indexDescriptor, final String properties) {
+
+    try (final var propertiesStream = IOUtils.toInputStream(properties, StandardCharsets.UTF_8)) {
+      return new PutMappingRequest.Builder()
+          .index(indexDescriptor.getIndexName())
+          .properties(deserializeJson(TypeMapping._DESERIALIZER, propertiesStream).properties())
+          .build();
+    } catch (final IOException e) {
+      throw new RuntimeException(
+          "Failed to load properties json into stream for put mapping into indexes matching the descriptor: ["
+              + indexDescriptor
+              + "]",
+          e);
+    }
+  }
+
+  private PutComponentTemplateRequest putComponentTemplateRequest(
+      final ComponentTemplateDescriptor templateDescriptor) {
+    try (final var template =
+        getClass()
+            .getClassLoader()
```

Comment:
💭 Looking at this again now, I'm thinking that `Thread.currentThread().getContextClassLoader()` is a better fit here. But we can tackle that later